pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -201,6 +202,96 @@ func TestAlphaDualStack(t *testing.T) {
 	}
 }
 
+func TestExternalInternalSubnet(t *testing.T) {
+
+	var subnetCidrINIConfig = `
+	[VirtualCenter "0.0.0.0"]
+	user = user
+	password = password
+	ip-family = ipv6,ipv4
+	[Nodes]
+	internal-network-subnet-cidr = "10.0.0.0/24"
+	external-network-subnet-cidr = "fd01:0:101:2609:bdd2:ee20:7bd7:0/112"
+	`
+
+	iniCfg, err := ccfg.ReadCPIConfigINI([]byte(subnetCidrINIConfig))
+	if err != nil{
+		t.Errorf("Seeing error %s when import INI config", err)
+	}
+	cfg, ok := configFromEnvOrSim(true)
+	defer ok()
+	iniCfg.Config = *cfg
+
+	connMgr := cm.NewConnectionManager(cfg, nil, nil)
+	defer connMgr.Logout()
+
+	ipv4Ip := "10.0.0.1"
+	ipv6Ip := "fd01:0:101:2609:bdd2:ee20:7bd7:5836"
+
+	nm := newNodeManager(iniCfg, connMgr)
+
+	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	vm.Guest.HostName = strings.ToLower(vm.Name) // simulator.SearchIndex.FindByDnsName matches against the guest.hostName property
+	vm.Guest.Net = []vimtypes.GuestNicInfo{
+		{
+			Network:   "foo-bar",
+			IpAddress: []string{ipv4Ip, ipv6Ip},
+		},
+	}
+
+	err = connMgr.Connect(context.Background(), connMgr.VsphereInstanceMap[cfg.Global.VCenterIP])
+	if err != nil {
+		t.Errorf("Failed to Connect to vSphere: %s", err)
+	}
+
+	// set config for ip for vc to ipv4, ipv6 (dual-stack)
+	vcInstance := nm.connectionManager.VsphereInstanceMap[cfg.Global.VCenterIP]
+	vcInstance.Cfg.IPFamilyPriority = []string{"ipv6", "ipv4"}
+
+	name := vm.Name
+	UUID := vm.Config.Uuid
+	k8sUUID := ConvertK8sUUIDtoNormal(UUID)
+
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Status: v1.NodeStatus{
+			NodeInfo: v1.NodeSystemInfo{
+				SystemUUID: k8sUUID,
+			},
+		},
+	}
+
+	// get node registered so node can be exported
+	nm.RegisterNode(node)
+	err = nm.DiscoverNode(name, cm.FindVMByName)
+	if err != nil {
+		t.Errorf("Failed DiscoverNode: %s", err)
+	}
+
+	nodeList := make([]*pb.Node, 0)
+	_ = nm.ExportNodes("", "", &nodeList)
+
+	ips := nodeList[0].Addresses
```

Comment:
Need to compare the AdressType in this test, to check if the right subnets were assigned InternalIP and ExternalIP

Reply:
There isn't a property where I can check the type of the ips. When the addresses are added, it's type agnostic as you can see [here.](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/nodemanager.go#L390)